### PR TITLE
IMAGEKIT-717

### PR DESCRIPTION
**Pull request by yashaswi-imagekit**
IMAGEKIT-717 : URL batch handling

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -112,7 +112,19 @@ class ImageKitUppyPlugin extends Plugin {
             if (file.error) {
                 return Promise.reject(new Error(file.error))
             } else {
-                return this.upload(file, current, total)
+                return new Promise((resolve, reject)=>{
+                    const queuedRequest = this.requests.run(()=>{
+                        this.upload(file, current, total)
```
Comment (niveditn) [inline src/main.js]: How is batch size limit being managed here?

File: src/main.js [hunk 2]
```diff
@@ -112,7 +112,19 @@ class ImageKitUppyPlugin extends Plugin {
...
+                        .then((res)=>{
+                            queuedRequest.done()
```
Comment (niveditn) [inline src/main.js]: Does `queuedRequest.done()` get called every time one file is uploaded? 

How do we know when an entire batch is finished, and when all the batches have finished, respectively?

**Comment by niveditn** [comment]
@yashaswi-imagekit Correct indentation of `.then` and `.catch` blocks w.r.t line 119, and bump package version to `1.0.4`

**Comment by yashaswi-imagekit** [comment]
@nikniv  done

**Review by niveditn — APPROVED** [review]

**Pull request merged**